api/c/indigo/src/indigo_savers.cpp

```diff
@@ -198,11 +199,23 @@ void IndigoSmilesSaver::generateSmarts(IndigoObject& obj, Array<char>& out_buffe
 
         SmilesSaver saver(output);
         saver.smarts_mode = true;
-
         if (mol.isQueryMolecule())
+        {
             saver.saveQueryMolecule(mol.asQueryMolecule());
+        }
         else
-            saver.saveMolecule(mol.asMolecule());
+        {
+            Array<char> mol_out_buffer;
+            ArrayOutput mol_output(mol_out_buffer);
+            MolfileSaver saver_tmp(mol_output);
+            saver_tmp.saveMolecule(mol.asMolecule());
+            mol_out_buffer.push(0);
+            BufferScanner sc(mol_out_buffer);
+            MolfileLoader loader_tmp(sc);
```

Comment:
Maybe it's more correct to use SMILES loader/saver in this case?

Reply:
Unfortunately SMILES parsers are buggy. We can't rely upon on it right now.